### PR TITLE
fix: fix create cron logic when create pipeline

### PR DESCRIPTION
**Pull request by Malyue**
#### What this PR does / why we need it:
create pipeline_cron and not create the pipeline_base when create pipeline

#### Which issue(s) this PR fixes:

- Fixes #your-issue_number
- [Erda Cloud Issue Link](https://erda.cloud/erda/dop/projects/387/issues/all?id=554193&iterationID=-1&type=TASK)

#### Specified Reviewers:

/assign @sfwn 

#### ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English |  fix create cron logic when create pipeline            |
| 🇨🇳 中文    |   修复新版流水线创建时会同时创建base记录问题，从创建base流程中单独提出cron创建流程来调用           |

#### Need cherry-pick to release versions?

Add comment like `/cherry-pick release/1.0` when this PR is merged.

> For details on the cherry pick process, see the [cherry pick requests](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md#how-to-cherry-pick-a-merged-pr) section under [CONTRIBUTING.md](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: internal/apps/dop/providers/projectpipeline/service_impl.go [hunk 1]
```diff
@@ -473,16 +488,52 @@ func (p *ProjectPipelineService) createCronIfNotExist(definition *dpb.PipelineDe
...
 
-	createV2 := extra.CreateRequest
-	createV2.DefinitionID = definition.ID
-	_, err = p.pipelineService.PipelineCreateV2(apis.WithInternalClientContext(context.Background(), discover.DOP()), createV2)
+	cronCreateRequest, err := p.constructCronCreateRequestFromV2(createV2, pipelineYml)
+	if err != nil {
+		return fmt.Errorf("failed to constructCronCreateRequestFromV2,err :%v", err)
+	}
+
+	_, err = p.PipelineCron.CronCreate(apis.WithInternalClientContext(context.Background(), discover.DOP()), cronCreateRequest)
```
Comment (sfwn) [inline internal/apps/dop/providers/projectpipeline/service_impl.go:496]: check if we need use context with user id

File: internal/apps/dop/providers/projectpipeline/service_impl.go [hunk 2]
```diff
@@ -473,16 +488,52 @@ func (p *ProjectPipelineService) createCronIfNotExist(definition *dpb.PipelineDe
...
 	if err != nil {
 		return fmt.Errorf("failed to CreatePipeline, err: %v", err)
 	}
 
 	return nil
 }
 
+// constructCronCreateRequestFromV2 make `PipelineCreateRequestV2` to `CronCreateRequest`
+func (p *ProjectPipelineService) constructCronCreateRequestFromV2(req *pipelinesvcpb.PipelineCreateRequestV2, pipelineYml *pipelineyml.PipelineYml) (*cronpb.CronCreateRequest, error) {
+	var err error
```
Comment (sfwn) [inline internal/apps/dop/providers/projectpipeline/service_impl.go:506]: move into `if pipelineYml == nil {` logic block

File: internal/tools/pipeline/providers/cron/cron.service.go [hunk 3]
```diff
@@ -46,6 +47,27 @@ func (s *provider) CronCreate(ctx context.Context, req *pb.CronCreateRequest) (*
...
+	if _, ok := req.FilterLabels[apistructs.LabelOwnerUserID]; !ok {
+		if req.OwnerUser != nil && req.OwnerUser.ID != nil {
+			ownerId := req.OwnerUser.ID.GetStringValue()
+			if ownerId == "" {
+				ownerId = conf.InternalUserID()
+			}
+			req.FilterLabels[apistructs.LabelOwnerUserID] = ownerId
+		}
+	}
+	if req.NormalLabels[apistructs.LabelOwnerUserID] != req.FilterLabels[apistructs.LabelOwnerUserID] {
```
Comment (sfwn) [inline internal/tools/pipeline/providers/cron/cron.service.go:67]: Confirm the purpose of owner id.

**Review by sfwn — CHANGES_REQUESTED** [review]

File: internal/apps/dop/providers/projectpipeline/service_impl.go (hunk 1 above)
Comment (Malyue) [reply internal/apps/dop/providers/projectpipeline/service_impl.go:496]: The `checkRolePermission` method is called before creation

File: internal/tools/pipeline/providers/cron/cron.service.go (hunk 3 above)
Comment (Malyue) [reply internal/tools/pipeline/providers/cron/cron.service.go:62]: The owner parameter is used to create a pipeline_extra record when starting a scheduled pipeline

File: internal/apps/dop/providers/projectpipeline/service_impl.go (hunk 2 above)
Comment (Malyue) [reply internal/apps/dop/providers/projectpipeline/service_impl.go]: has been fixed

File: api/proto/core/pipeline/cron/cron.proto [hunk 4]
```diff
@@ -100,6 +100,11 @@ message CronCreateRequest {
   google.protobuf.Timestamp cronStartFrom = 12;
   map<string,string> incomingSecrets = 13;
   string pipelineDefinitionID = 14;
+
+  common.IdentityInfo identityInfo = 15;
+  core.pipeline.base.PipelineUser ownerUser = 16;
```
[the remark this answers is not in the document]
Comment (Malyue) [reply api/proto/core/pipeline/cron/cron.proto:105]: the ownerUser is from gittar, not use the identityInfo.User

File: api/proto/core/pipeline/cron/cron.proto [hunk 5]
```diff
@@ -100,6 +100,11 @@ message CronCreateRequest {
...
+
+
```
[the remark this answers is not in the document]
Comment (Malyue) [reply api/proto/core/pipeline/cron/cron.proto]: has been fixed

File: api/proto/core/pipeline/cron/cron.proto [hunk 6]
```diff
@@ -86,7 +86,7 @@ message CronStopResponse {
 
 // cron create
 message CronCreateRequest {
-  uint64 ID = 1;
+  uint64 ID = 1; // TODO 改进一下逻辑，创建请求应该不需要传id
```
Comment (sfwn) [inline api/proto/core/pipeline/cron/cron.proto:89]: delete this comment

**Review by sfwn — CHANGES_REQUESTED** [review]

File: api/proto/core/pipeline/cron/cron.proto (hunk 6 above)
Comment (Malyue) [reply api/proto/core/pipeline/cron/cron.proto]: has been fixed

**Review by sfwn — APPROVED** [review]

**Pull request merged**